### PR TITLE
Report API + Report page improvements

### PR DESCRIPTION
**Pull request by pouriaMaleki**

### CODE REVIEW

File: backend/src/feedback/report.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  parent,
+  { reportUid },
+  { prisma },
+) => {
+  const feedbackBase = await prisma.feedbackBase.findUnique({
+    where: { reportUid: reportUid },
+    include: { feedbackResponses: true },
+  });
+
+  return feedbackBase as FeedbackBase & { feedbackResponses: FeedbackResponse[] };
```
Comment (pouriaMaleki) [inline backend/src/feedback/report.ts:20]: Couldn't make this right :(

File: backend/src/feedback/report.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import {
+  FeedbackBase,
+  FeedbackResponse,
+  QueryResolvers,
+} from '../graphql/resolvers-types';
+import { makeFeedbackLink, makeReportLink } from './utils/makeLinks';
+
+export const DAY = 24 * 60; // a day is 24h * 60min
+export const GENERATE_LINK_DAY_LIMIT = 3;
```
Comment (albacoretuna) [inline backend/src/feedback/report.ts:9]: I couldn't figure out where GENERATE_LINK_DAY_LIMIT was used.

File: backend/src/graphql/resolvers-types.ts [hunk 3]
```diff
@@ -28,22 +29,41 @@ export type Response = {
...
+  id: Scalars['Int'];
   title?: Maybe<Scalars['String']>;
-  createdAt?: Maybe<Scalars['String']>;
+  feedbackUid: Scalars['String'];
+  reportUid: Scalars['String'];
+  feedbackLink: Scalars['String'];
+  reportLink: Scalars['String'];
+  createdAt: Scalars['DateTime'];
+  updatedAt: Scalars['DateTime'];
+  feedbackResponses: Array<FeedbackResponse>;
```
Comment (albacoretuna) [inline backend/src/graphql/resolvers-types.ts:54]: hmmmm just as a side note. I see a cycle here. Feedbackbase and FeedbackResponse include each other. Is this circular dependence ok? Probably it is :)

**Review by albacoretuna — APPROVED** [review]
Well done!

File: backend/src/feedback/report.ts (hunk 2 above)
Comment (pouriaMaleki) [reply backend/src/feedback/report.ts:9]: Just some copy-paste shit :D

File: backend/src/graphql/resolvers-types.ts (hunk 3 above)
Comment (pouriaMaleki) [reply backend/src/graphql/resolvers-types.ts:54]: Yeah, actually it is possible to get the Feedbackbase from the FeedbackResponse and vice versa in the Graphql queries. We can limit the levels though. I think we might actually have this in the backlog already.

**Pull request merged**